test/e2e/framework/metrics/metrics_grabber.go

```diff
@@ -63,8 +62,16 @@ type Grabber struct {
 	waitForControllerManagerReadyOnce sync.Once
 }
 
+var setupProxyOnce sync.Once
```

Comment:
Beware that this won't synchronize across different Gingko workers: each worker is a different process, so tests being executed in different workers in parallel may end up calling `setupMetricsProxy` in parallel.

Reply:
I'm not aware of any way to do initialization only once depending on which tests are going to run.

Reply:
Oops.. I didn't realize that. Does it mean we could only do initialization in `setupSuite`? cc @aojea